### PR TITLE
MES-2744 RSIS mapping for adi pt2

### PR DESCRIPTION
**Pull request by andrewsetterfield**
# Description and relevant Jira numbers

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

**Comment by andrewsetterfield** [comment]
json test data and mapped object from db attached
[adi2 mapping.xlsx](https://github.com/dvsa/mes-mi-export-service/files/4481192/adi2.mapping.xlsx)

File: src/functions/uploadToRSIS/application/data-mapper/__tests__/data-mapper.spec.ts [hunk 1]
```diff
@@ -26,6 +27,14 @@ import {
...
 import { getFullyPopulatedDrivingFaults } from './helpers/cat-a-mod2/inputs/fully-populated-inputs';
 import { getCatAMod2MinimalInput } from './helpers/cat-a-mod2/inputs/minimal-inputs';
+import { getCatADI2MinimalInput } from './helpers/cat-adi2/inputs/minimal-inputs';
+import {
+  getADI2FullyPopulatedDrivingFaults,
+  getADI2FullyPopulatedSeriousFaults,
+  getADIFullyPopulatedDangerousFaults,
+} from './helpers/cat-adi2/inputs/fully-populated-inputs';
+import { CatADI2UniqueTypes } from '@dvsa/mes-test-schema/categories/ADI2';
+import TestData = CatADI2UniqueTypes.TestData;
```
Comment (matthew2564) [inline src/functions/uploadToRSIS/application/data-mapper/__tests__/data-mapper.spec.ts]: Not sure if going mad but don't think this is the the typical way to import TestData

File: src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+  optionalBoolean,
+  formatManoeuvreDangerous,
+  addIfSet,
+  getCompetencyComments,
+  formatQuestionFaultADI2,
+  formatQuestionCompleted,
+} from './data-mapper';
+import { formatGearboxCategory } from '../helpers/shared-formatters';
+import { CatADI2UniqueTypes } from '@dvsa/mes-test-schema/categories/ADI2';
+import TestData = CatADI2UniqueTypes.TestData;
```
Comment (matthew2564) [inline src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts]: same as above in reference to not the typical way of importing TestData

File: src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+/**
+ * Maps data specific to the ``ADI2`` test category.
+ *
+ * @param result The MES test result
+ * @returns The mapped MI Export data
+ * @throws MissingTestResultDataError If mandatory data missing from MES test result
+ */
+export const mapCatADI2Data = (result: ResultUpload): DataField[] => {
+  const t = result.testResult.testData;
```
Comment (matthew2564) [inline src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts]: if we change this to be:
  const t = result.testResult.testData as TestData;
should't then have to have:
t as TestData throughout, can just have t

Comment (andrewsetterfield) [reply src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts]: agreed. updated.

File: src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts (hunk 2 above)
Comment (andrewsetterfield) [reply src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts]: TestData isn't exported directly, so can only reference it via CatADI2UniqueTypes.

File: src/functions/uploadToRSIS/application/data-mapper/__tests__/data-mapper.spec.ts (hunk 1 above)
Comment (andrewsetterfield) [reply src/functions/uploadToRSIS/application/data-mapper/__tests__/data-mapper.spec.ts]: TestData isn't exported directly, so can only reference it via CatADI2UniqueTypes.

File: src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts (hunk 2 above)
Comment (matthew2564) [reply src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts]: is it not this?

import { TestData } from '@dvsa/mes-test-schema/categories/ADI2/partial';

File: src/functions/uploadToRSIS/application/data-mapper/__tests__/data-mapper.spec.ts (hunk 1 above)
Comment (andrewsetterfield) [reply src/functions/uploadToRSIS/application/data-mapper/__tests__/data-mapper.spec.ts]: removed import of test data and am accessing where needed directly from CatADI2UniqueTypes

File: src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts (hunk 2 above)
Comment (andrewsetterfield) [reply src/functions/uploadToRSIS/application/data-mapper/cat-adi2-mapper.ts]: removed import of test data and am accessing where needed directly from CatADI2UniqueTypes

**Review by tomcrawleyy — APPROVED** [review]
looks good

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**